src/parser/ParsedQuery.h

```diff
@@ -307,13 +307,25 @@ class ParsedQuery {
     }
   };
 
+  using Asterisk = char;
+  using _selectedVariables = vector<string>;
+  /*
+    struct SelectAsterisk {
+    char _asterisk = '*';
+    bool _reduced = false;
+    bool _distinct = false;
+  };
+  */
+
```

Comment:
Use something like
```
 struct SelectVarsOrAsterisk {
   private:
    std::variant<_selectedVariables,Asterisk> _varsOrAsterisk;
   public:
    bool isAsterisk() const {
      return std::holds_alternative<char>(_varsOrAsterisk);
    }
    const auto& getSelectVariables() const {
      return std::get<std::vector<string>>(_varsOrAsterisk);
    }
    auto& getSelectVariables()  {
      return std::get<std::vector<string>>(_varsOrAsterisk);
    }
  };
```
And then the `SelectClause` has a member of this type,
and the rest of the code gets more readable.